test/parallel/market.rpc.check.test.ts

```diff
@@ -0,0 +1,473 @@
+          threshold.muln(10),
+          secondCurrency,
+          threshold.muln(5),
+          poolType,
+        ),
+        Assets.mintNative(testUser),
+        Assets.mintToken(firstCurrency, testUser, threshold.muln(20)),
+      );
+
+      await updateFeeLockMetadata(sudo, null, null, null, [
+        [firstCurrency, true],
+      ]);
+
+      liqId = await getPoolId(firstCurrency, secondCurrency);
+    });
+
+    test("Function calculateSellPriceByMarket works correctly", async () => {
+      const sellPrice = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold.muln(2),
+      );
+
+      if (poolType === "Xyk") {
+        const poolBalance = await getBalanceOfPool(
+          firstCurrency,
+          secondCurrency,
+        );
+        priceCalculationRetro = await calculate_sell_price_rpc(
+          poolBalance[0],
+          poolBalance[1],
+          threshold.muln(2),
+        );
+      } else {
+        priceCalculationRetro = sellPrice;
+      }
+
+      //we create reverse calculation to check that we receive another value
+      const sellPriceReverse = await calculateSellPriceByMarket(
+        liqId,
+        secondCurrency,
+        threshold.muln(2),
+      );
+
+      await signTx(
+        api,
+        Market.sellAsset(
+          liqId,
+          firstCurrency,
+          secondCurrency,
+          threshold.muln(2),
+        ),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const tokenAmount = await getTokensAccountInfo(
+        testUser.keyRingPair.address,
+        secondCurrency,
+      );
+
+      expect(sellPrice).bnEqual(priceCalculationRetro);
+      expect(sellPrice).not.bnEqual(sellPriceReverse);
+      expect(stringToBN(tokenAmount.free)).bnEqual(sellPrice);
+    });
+
+    test("Function calculateSellPriceWithImpact makes reliable calculations", async () => {
+      const sellPriceBefore = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      const sellPriceWithImpact = await calculateSellPriceWithImpact(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      await signTx(
+        api,
+        Market.sellAsset(liqId, firstCurrency, secondCurrency, threshold),
+        testUser.keyRingPair,
+      ).then((result) => {
+        const eventResponse = getEventResultFromMangataTx(result);
+        expect(eventResponse.state).toEqual(ExtrinsicResult.ExtrinsicSuccess);
+      });
+
+      const sellPriceAfter = await calculateSellPriceByMarket(
+        liqId,
+        firstCurrency,
+        threshold,
+      );
+
+      const diff = sellPriceAfter.sub(sellPriceWithImpact[1]);
+
+      expect(sellPriceBefore).bnEqual(sellPriceWithImpact[0]);
+      expect(diff).bnLt(sellPriceAfter.muln(3).divn(1000));
```

Comment:
do not fully understand this guy, can u plz add a comment about it?

Reply:
I discussed it with Marian. Function rpc.market.calculate_sell_price  is the function that was created for FE, and it needs to predict what result we received if we repeat our operation twice. But at the same time, we have a slight discrepancy in the region of 3% for the second calculation. I consider the difference between the second calculation of the function and the result obtained, and if it is less than 3%, then I consider the test successful.